src/game/AI/ScriptDevAI/scripts/kalimdor/dire_maul/dire_maul.cpp

```diff
@@ -170,6 +235,13 @@ struct DreadsteedQuestObjects : public GameObjectAI
 
     void JustSpawned() override
     {
+        if (instance_dire_maul* instance = (instance_dire_maul*)m_go->GetInstanceData()) // Without this check, the 3 objects are spawning when first loaded in. Couldn't find another way to stop them from doing this
```

Comment:
Old spawning system, if migrated to spawn groups OR spawnmask 0, this issue doesnt occur. One of the reasons I had to move away from how it used to work.